physionet-django/project/modelcomponents/activeproject.py

```diff
@@ -295,12 +280,24 @@ def check_integrity(self):
             if not author.affiliations.all():
                 self.integrity_errors.append('Author {0} has not filled in affiliations'.format(author.user.username))
 
+        # Content
+        sections = ProjectSection.objects.filter(
+            resource_type=self.resource_type, required=True)
+        for attr in sections:
+            try:
+                section = self.project_content.get(project_section=attr)
+                if not section.is_valid():
+                    raise ActiveSectionContent.DoesNotExist
+            except ActiveSectionContent.DoesNotExist:
+                self.integrity_errors.append(
+                    'Missing required field: {0}'.format(attr.title))
```

Comment:
The amount of database queries could also be reduced.

Reply:
Done.